### ISSUE TITLE
error "out of memory"

### ISSUE THREAD

**Issue by tulikasriv26** [vrmarcelino/CCMetagen#66]
I am using ccmetagen for so long. But now I am using it since three months and this error is displayed for every sample. I have tried using single sample too for analysis and asked for memory 500 gb, 800gb ; but I am receiving the similar error after 2 hours of run. This is the first time I am receiving errors in this way. I am using batch script at compute canada cluster. 
#!/bin/bash
#SBATCH --nodes=1
#SBATCH --mem=500G
#SBATCH --account=#######
#SBATCH --cpus-per-task 30
#SBATCH -J ccmet_single
#SBATCH -o ccmet_single.out
#SBATCH -e ccmet_single.err
#SBATCH --time=24:00:00
#SBATCH --mail-user=##########
#SBATCH --mail-type=ALL
module load kma/1.4.14
kma -ipe X0B1_MA0919_I_000162_80C_11.qc.R1.fastq.gz X0B1_MA0919_I_000162_80C_11.qc.R2.fastq.gz -o X0B1_MA0919_I_000162_80C_11_out_kma -t_db /home/tuli15/scratch/compress_ncbi_nt/ncbi_nt -t 30 -1t1 -mem_mode -and -apm f

**Comment by vrmarcelino** [comment]
Hi!
Did you change something from the previous analysis that worked (kma version or database?)
Could you also try reducing the number of threads to 6?
Thanks!

**Comment by tulikasriv26** [comment]
No, I haven't changed the database. I was performing analysis for 80 samples. 76 samples were completely analyzed, only 4 samples are showing this error. I have checked remaining samples with other tools too like kraken2, their analysis is perfect with them. 
And I have reduced threads to 8,6 and 4 repeatively but they are showing the similar error. I am utilizing ncbi database. Can you please suggest something? And I have tried analysis on two other clusters too .

**Comment by vrmarcelino** [comment]
Interesting - if it worked for some samples it means that there might be an issue with these 4 files, or actual memory, or space.

Could just actually be out of memory (files are larger than usual?) or could you be out of temporary space? 
You can try adding the flag: `-tmp <dir_where_you_have_plenty_of_space>` ?

Does the analysis start and takes a while before crashing? We had weird error messages with incomplete files (or files with different number of forward and reverse reads for example), which sometimes work with other classifiers.

**Comment by tulikasriv26** [comment]
I have tried other analysis with remaining 4 files . They are working just fine but not with ccmetagen. Yes, the analysis will run for around 2 hours and suddenly shows me exit code 122 ("run_out_of_memory"). 
To be very honest. I have 20 TB space available in my working directory. It is so weird for me for the first time with CCMetagen

**Comment by vrmarcelino** [comment]
Did you try specifying the temporary directory with -tmp ?
Temporary files may be saved in a different place from your current directory.

**Comment by tulikasriv26** [comment]
Yes I do, even I have removed every single data from the -tmp directory and the space shows around 10TB. I have no idea honestly what is happening. Just to run one sample and for output files I have 10 TB ; it is something new error to me.

**Comment by vrmarcelino** [comment]
Other things to try:
Try running in a subset of the file (can you confirm that you don't have an extra space in the beginning of the fasta file?)
Remove the -tmp flag (I know it is weird, but worked once for us).
Try a different version of kma, and/or a different HPC computer.
You can also try to place this issue in the KMA website.
